### ISSUE TITLE
Intermediate code generation with generics: type argument ... is not within bounds of type-variable T

### ISSUE THREAD

**Issue by hosswald** [stoicflame/enunciate#775]
I'm trying to generate an API documentation using enunciate. One of my REST routes returns a `ServerAnswer<Subscription>`, where `ServerAnswer` is one of our classes and `com.stripe.model Subscription extends APIResource implements MetadataStore<Subscription>, HasId` is a [Stripe API](https://stripe.com/docs/api) class.

When running the Gradle `enunciate` task, I get a bunch of the following Java compiler errors:
```
C:\Users\...\build\enunciate\java-json-client\src\com\stripe\model\CustomerSubscriptionCollection.java:15: error: type argument Subscription is not within bounds of type-variable T
public class CustomerSubscriptionCollection extends com.stripe.model.StripeCollection<com.stripe.model.Subscription> {
                                                                                                      ^
  where T is a type-variable:
    T extends HasId declared in class StripeCollection
```
I get this message for a number of other classes in ´com.stripe.model.*´ referencing each other.

Looking at the files generated by enunciate, the reason is obvious:
```
@JsonInclude( JsonInclude.Include.NON_NULL )
@JsonIgnoreProperties( ignoreUnknown = true )
public class CustomerSubscriptionCollection extends com.stripe.model.StripeCollection<com.stripe.model.Subscription> {...}
```
```
@JsonInclude( JsonInclude.Include.NON_NULL )
@JsonIgnoreProperties( ignoreUnknown = true )
public class StripeCollection<T extends com.stripe.model.HasId> extends com.stripe.model.StripeObject {...}
```
```
@JsonInclude( JsonInclude.Include.NON_NULL )
@JsonIgnoreProperties( ignoreUnknown = true )
public class Subscription extends com.stripe.net.APIResource {...}
```
`StripeCollection`s require type arguments extending `HasId`, but the generated `Subscription` class does not extend it.
Whereas the original class does:
```
public class Subscription extends APIResource implements MetadataStore<Subscription>, HasId {...}
```
(from [stripe-java-3.11.0-sources.jar](http://repo1.maven.org/maven2/com/stripe/stripe-java/3.11.0/)).
I'm guessing the enunciate code generator omits the extension of `HasId`, since it is an interface and thus not relevant to class fields. But it does not do so in all cases (namely in the generated StripeCollection.java).

**Comment by stoicflame** [comment]
Fixed at c475d41. Verified with [this patch](https://github.com/stoicflame/enunciate/files/1948846/enunciate-775.diff.txt) to the [enunciate-sample](https://github.com/stoicflame/enunciate-sample) app.

**Comment by stoicflame** [comment]
[Enunciate 2.11 has been released](https://github.com/stoicflame/enunciate/releases/tag/v2.11.0).
